Create template tag to render volunteer languages

ArnavBallinCode opened the pull request:
- Created volunteer_tags.py with two template tags:
  - volunteer_languages_badges: inclusion tag for rendering languages
  - render_volunteer_languages: simple tag for inline rendering
- Refactored volunteerprofile_detail.html to use template tag
- Refactored volunteerprofile_review_form.html to use template tag
- Added comprehensive tests for template tags
- Added documentation for the new template tags
- Eliminates code duplication for volunteer language rendering

Fixes #154 , work in progress, testing properly...

ArnavBallinCode committed:
Create template tag to render volunteer languages

- Created volunteer_tags.py with two template tags:
  - volunteer_languages_badges: inclusion tag for rendering languages
  - render_volunteer_languages: simple tag for inline rendering
- Refactored volunteerprofile_detail.html to use template tag
- Refactored volunteerprofile_review_form.html to use template tag
- Added comprehensive tests for template tags
- Added documentation for the new template tags
- Eliminates code duplication for volunteer language rendering

Fixes #154 , work in progress, testing properly...

ArnavBallinCode committed:
Fix code formatting for volunteer template tags

- Apply black formatting to volunteer_tags.py and test_templatetags.py
- Fix import order with isort (Django imports alphabetically sorted)
- Ensure consistent code style compliance for CI

Mariatta commented on docs/developer/volunteer_templatetags.md:
Also the file needs to be added to the nav in mkdocs.yml file

ArnavBallinCode committed:
Address reviewer feedback: Rename and restructure template tags documentation

- Rename volunteer_templatetags.md to templatetags.md for better organization
- Restructure content to support future template tag categories (sponsorship, portal, etc.)
- Add templatetags.md to mkdocs.yml navigation under Developer Guide
- Update heading structure for better organization and readability

ArnavBallinCode committed:
Merge branch 'main' into feature/volunteer-languages-templatetag

ArnavBallinCode committed:
Merge branch 'main' into feature/volunteer-languages-templatetag

ArnavBallinCode committed:
Merge branch 'main' into feature/volunteer-languages-templatetag

ArnavBallinCode committed:
Merge branch 'main' into feature/volunteer-languages-templatetag
